src/swap/FeeInfoBottomSheet.tsx

```diff
@@ -0,0 +1,199 @@
+import React from 'react'
+import { Trans, useTranslation } from 'react-i18next'
+import { StyleSheet, Text, View } from 'react-native'
+import BottomSheet, { BottomSheetRefType } from 'src/components/BottomSheet'
+import Button, { BtnSizes, BtnTypes } from 'src/components/Button'
+import TokenDisplay from 'src/components/TokenDisplay'
+import Colors from 'src/styles/colors'
+import { typeScale } from 'src/styles/fonts'
+import { Spacing } from 'src/styles/styles'
+import { AppFeeAmount, SwapFeeAmount } from 'src/swap/types'
+
+interface Props {
+  forwardedRef: React.RefObject<BottomSheetRefType>
+  appFee?: AppFeeAmount
+  crossChainFee?: SwapFeeAmount
+  networkFee?: SwapFeeAmount
+  fetchingSwapQuote: boolean
+}
+
+function FeeAmount({
+  fee,
+  showMaxAmount,
+  isLoading,
+}: {
+  fee: SwapFeeAmount | undefined
+  showMaxAmount?: boolean
+  isLoading: boolean
+}) {
+  const { t } = useTranslation()
+
+  const displayAmount = showMaxAmount ? fee?.maxAmount : fee?.amount
+  const isFeeZero = displayAmount && displayAmount.isZero()
+  const loadingText = t('loading')
+  const fallbackText = t('unknown')
+  const zeroFeeText = t('swapScreen.transactionDetails.swapFeeWaived')
+
+  return (
+    <Text style={styles.feeAmountText}>
+      {isLoading ? (
+        loadingText
+      ) : isFeeZero ? (
+        zeroFeeText
+      ) : fee && fee.token && displayAmount ? (
+        <>
+          {`≈ `}
+          {fee.token.priceUsd ? (
+            <>
+              <TokenDisplay
+                amount={displayAmount}
+                tokenId={fee.token.tokenId}
+                style={styles.bodyText}
+                errorFallback=""
+              />
+              {` (`}
+              <TokenDisplay
+                amount={displayAmount}
+                tokenId={fee.token.tokenId}
+                showLocalAmount={false}
+                style={styles.bodyText}
+                errorFallback=""
+              />
+              {`)`}
```

Comment:
We're probably copy/pasting this, but we should avoid concatenating strings, it makes localization harder.
We can use a `<Trans>` component.

Reply:
updated!